src/index.tsx

```diff
@@ -209,10 +211,14 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
     public render() {
         const {
             trigger,
+            onBeforeGetContent,
         } = this.props;
 
         return React.cloneElement(trigger(), {
-            onClick: this.handlePrint,
+            onClick: () => onBeforeGetContent ? onBeforeGetContent().then((e) => {
```

Comment:
This will require a check similar to [here](https://github.com/gregnb/react-to-print/blob/67df094343115d35f002ef66a752abb3a6b9d261/src/index.tsx#L57) so that we can handle people who do not have browsers that support Promise.

Reply:
you're right

I'll improve to this.
Thank you :)